src/main/java/com/google/firebase/messaging/AndroidFcmOptions.java

```diff
@@ -0,0 +1,67 @@
+package com.google.firebase.messaging;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.api.client.util.Key;
+import com.google.firebase.internal.FcmOptionsUtil;
+
+/**
+ * Represents the Android-specific FCM options that can be included in an {@link AndroidConfig}.
+ * Instances of this class are thread-safe and immutable.
+ */
+public class AndroidFcmOptions {
+
+  @Key("analytics_label")
+  private final String analyticsLabel;
+
+  private AndroidFcmOptions(Builder builder) {
+    checkArgument(FcmOptionsUtil.validAnalyticsLabelFormat(builder.analyticsLabel),
+        "Analytics label must have format matching'^[a-zA-Z0-9-_.~%]{1,50}$'");
+    this.analyticsLabel = builder.analyticsLabel;
+  }
+
+  /**
+   * Creates a new {@link AndroidFcmOptions} with the specified analytics label.
+   *
+   * @param analyticsLabel An analytics label
+   * @return An AndroidFcmOptions with the analytics label set to the supplied value.
+   */
+  public static AndroidFcmOptions withAnalyticsLabel(String analyticsLabel) {
+    return new AndroidFcmOptions(new Builder().setAnalyticsLabel(analyticsLabel));
```

Comment:
Nit: Use the public API here. We should minimize direct invocations of the private APIs.

```
return builder().setAnalyticsLabel(analyticsLabel).build();
```

Reply:
done